BankManagerConsoleApp/BankManagerConsoleTests/QueueTests/QueueTests.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  QueueTests.swift
+//  QueueTests
+//
+//  Created by Dasan & Mary on 2023/07/10.
+//
+
+import XCTest
+@testable import BankManagerConsoleApp
+
+final class QueueTests: XCTestCase {
+    var sut: Queue<Int>!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        
+        sut = Queue()
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+        
+        sut = nil
+    }
+    
+    func test_enqueue_enqueue_1을했을때_queue의_peek값은_1이다() {
+        // given
+        sut.enqueue(1)
+        let expectation = 1
+        
+        // when
+        let result = sut.peek
+        
+        // then
+        XCTAssertEqual(result, expectation)
+    }
+    
+    func test_enqueue_enqueue_3_4_5를했을때_queue의_peek값은_3이다() {
+        // given
+        sut.enqueue(3)
+        sut.enqueue(4)
+        sut.enqueue(5)
+        let expectation = 3
+        
+        // when
+        let result = sut.peek
+        
+        // then
+        XCTAssertEqual(result, expectation)
+    }
+    
+    func test_dequeue_조건없이_queue에_dequeue를하면_nil를_반환한다() {
+        // given, when
+        let result = sut.dequeue()
+        
+        // then
+        XCTAssertNil(result)
+    }
+    
+    func test_dequeue_enqueue_3_4_5를하고_dequeue를하면_3을_반환한다() {
+        // given
+        sut.enqueue(3)
+        sut.enqueue(4)
+        sut.enqueue(5)
+        let expectation = 3
+ 
+        // when
+        let result = sut.dequeue()
+
+        // then
+        XCTAssertEqual(result, expectation)
+    }
+    
+    func test_dequeue_enqueue_3_4_5를하고_dequeue를_3번했을때_3_4_5를_순서대로_반환한다() {
+        // given
+        sut.enqueue(3)
+        sut.enqueue(4)
+        sut.enqueue(5)
+        let expectation = [3, 4, 5]
+        
+        // when
+        var result: [Int?] = [Int?]()
+        for _ in 1...3 {
+            result.append(sut.dequeue())
+        }
```

Comment:
이러면 관리 포인트가 많아지니까 아래와 같이 바꾸는건 어떨까요?(expectation이 변경되면 수동으로 enqueue, dequeue를 맞춰줘야 하니까)
```suggestion
        let expectation = [3, 4, 5]
        expectation.forEach(sut.enqueue)
        
        // when
        var result: [Int?] = (0..<expectation.count).map { _ in sut.dequeue() }
```

Reply:
이렇게 하면 `enqueue`와 `dequeue`를 신경쓰지 않고 `expectation`만 변경해도 되는군요! 
수정 완료하였습니다:)